Pokemon Unity/Assets/Scripts2/Pokemon/Pokemon.cs

```diff
@@ -269,6 +265,70 @@ public Pokemon(Pokemons pokemon) : this()
 
             //calcStats();
         }
+
+        public Pokemon(string nickName, int form,
+            Pokemons species, Abilities ability,
+            Nature nature,
+            bool isShiny, bool? gender,
+            int[] pokerus, int pokerusStrain,
+            int? shadowLevel,
+            int currentHp, Items item,
+            byte[] iv, byte[] ev, 
+            int obtainedLevel, int currentLevel, int currentExp,
+            int happiness, Status status, int statusCount,
+            int eggSteps, Items ballUsed,
+            Item.Item.Mail mail, Move[] moves,
+            Ribbon[] ribbons, bool[] markings,
+            int personalId,
+            ObtainedMethod obtainedMethod,
+            DateTimeOffset timeReceived, DateTimeOffset timeEggHatched) : this()
+        {
+            name = nickName;
```

Comment:
"IF" nickname isnt null... Some pokemons have default names, and no nicks assigned. Inside the pokemon's class, it checks to see if the nickname is null before spitting out string information. This might be an issue, if you're adding values when none exist.

Reply:
Does it work on string.Empty? Or does it do a null check?

Reply:
Both, i think. There is the existence of `string.IsNullOrEmpty` and `string.IsNullOrWhiteSpace`. I think I used the second. So if a name is all spacebar, it will be ignored and reverted back to "Pokemon.Name"

Reply:
if (nickName != null || nickName != string.Empty)
{
                name = nickName;
}
else
{
                name = null;
}

This is the null check I did for the new constructor now.